### ISSUE TITLE
sonarScan failing with multi module not directly under root path

### ISSUE THREAD

**Issue by joelvim** [sonar-scala/sbt-sonar#175]
I have a project with many multiple modules. These modules are organised under subfolders like this:
```
my-project-root
  src
    main
      scala
  subfolder-1
    module-1
      src
        main
          scala
  subfolder-2
    module-2
      src
        main
          scala
```
with the main module directly under root directory, a module under `subfolder-1/module-1` and a second module under `subfolder-2/module-2`.

sonarScan fails in this configuration : 
- if I do not declare any submodule of the main module (consider they are independent), it fails because it searches sources under `module-1` instead of `subfolder-1/module-1`
- if I add `sonar.projectBaseDir` on each module, but do not declare them as submodules, it works
- if I add `sonar.projectBaseDir` and declare submodules, it fails.

It seems it needs the `sonar.projectBaseDir` under each module, and in case of module aggregation under the main module, it needs a `module1.sonar.projectBaseDir` for each module.

I will provide a PR putting this in evidence.

**Comment by joelvim** [comment]
If we configure each module `projectBaseDir` on main module, it works.
It would be awesome if there was a boolean setting that allowed user to choose if he want this to be done automatically, because doing this by hand is not maintainable when you have a lot of modules (50 in my case)

However, I don't know how, with sbt API, you can get, for a project, all its aggregated project, and get their id and base path in order to build the correct sonar configuration :'(

**Comment by joelvim** [comment]
> Out of curiosity, are you using the official SonarQube's SonarScala plugin or our community driven sonar-scala? I remember there were some issues with using the base dir setting in sonar-scala, but I can't quite remember where I left this - I'll check that.

I was testing in local with `mwizner/sonarqube-scala-plugins` docker image, but the deployed sonar is an old 6.7 community with only the standard scala plugin. I will later see if we can upgrade and use sonar-scala plugin.

> That's quite an unconventional setup you have there with modules in subdirectories, is there any particular reason why you have your project set up that way?

The original reason was because we have > 50 modules in our repo, and it was a way to avoid having all with directories at root, so it was easier to work with it.

> Yes, I agree with you that it would be useful to populate sonar.modules and moduleName.sonar.projectName automatically, maybe it's something that could be automated. 

I spent some time trying to do it but I didn't succeed. Even when I put all my modules in a list and tried to transform them as the correct sonar properties, I encountered sbt errors that were away of my sbt dev knowledge :smile:

**Comment by mwz** [comment]
Thanks again for the PR, I've merged the changes and will release a new version shortly.

Regarding populating properties for all modules automatically I think there is a way in sbt to get a list of projects which are aggregated by the current project, but I'm not sure how to go from a project reference to getting settings of that project; worth asking on sbt's gitter channel, maybe someone can point you in the right direction.

**Comment by mwz** [comment]
Released as v2.3.0, should be in maven central shortly.

**Comment by joelvim** [comment]
Thank you very much for the merge and the release. I didn't think about asking on the gitter channel, maybe it could help.
